apps/web/app/modules/Admin/EditCourse/CourseLessons/NewLesson/QuizLessonForm/components/AnswerSelectQuestion.tsx

```diff
@@ -0,0 +1,172 @@
+import * as Accordion from "@radix-ui/react-accordion";
+import { Label } from "@radix-ui/react-label";
+import { useCallback, useMemo, useState } from "react";
+
+import { Icon } from "~/components/Icon";
+import { Button } from "~/components/ui/button";
+import { Input } from "~/components/ui/input";
+
+import QuestionTitle from "./QuestionTitle";
+
+import type { QuizLessonFormValues } from "../validators/quizLessonFormSchema";
+import type { UseFormReturn } from "react-hook-form";
+import { QuestionOption, QuestionType } from "../QuizLessonForm.types";
+import { cn } from "~/lib/utils";
+
+type AnswerSelectQuestionProps = {
+  form: UseFormReturn<QuizLessonFormValues>;
+  questionIndex: number;
+};
+
+const AnswerSelectQuestion = ({ form, questionIndex }: AnswerSelectQuestionProps) => {
```

Comment:
tip for the future - you can use `<FormProvider />` with `useFormContext()` hooks instead drilling entire `form` object through props 😉